### DISCUSSION TITLE
Task 5: Implement Uniswap V3 python refactor into uniswappy

### DISCUSSION THREAD

**Discussion by icmoore** [defipy-devs/defipy#11] [Ideas]
This task of implementing Uniswap V3 has demand from SYS Labs and from the general community at large. This would be very research intensive and would require: (1) deep understanding of Uniswap v3 math and (2) some intermediate solidity experience.

As far as I know, there are no trusted OSS python refactors of Uniswap v3; however there are some places to start digging to get ideas

- https://github.com/tradingstrategy-ai/web3-ethereum-defi
- https://github.com/chainflip-io/chainflip-uniswapV3-python
- https://github.com/DefiLab-xyz/uniswap-v3-backtest-python
- https://github.com/aloelabs/uniswap-simulator

**Comment by icmoore** [comment]
I will be taking on this task, ideas and suggestions are welcome

**Comment by Andrei-1234** [comment]
Ok!

**Reply by icmoore** [reply]
Pre-release is ready for use on main repos!

**Comment by icmoore** [comment]
Now available >= v.1.3.0
